view/frontend/templates/button_product_page.phtml

```diff
@@ -0,0 +1,278 @@
+<?php
+/**
+ * Bolt magento2 plugin
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/osl-3.0.php
+ *
+ * @category   Bolt
+ * @package    Bolt_Boltpay
+ * @copyright  Copyright (c) 2018 Bolt Financial, Inc (https://www.bolt.com)
+ * @license    http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+
+/**
+ * Product page Bolt Checkout button
+ *
+ * @var $block \Bolt\Boltpay\Block\JsProductPage
+ */
+if ($block->shouldDisableBoltCheckout()) return;
+
+if (!$block->isSupportableType()) return;
+
+$trackCallbacks = $block->getTrackCallbacks();
+
+?>
+    <div class="bolt-product-checkout-button bolt-multi-step-checkout <?=$block->getAdditionalCheckoutButtonClass();?>"></div>
+    <script>
+        // Store the configuration parameters passed from the php block
+        // in the global object. Used in this file and on the payment page
+        // in payment method renderer, vendor/boltpay/bolt-magento2/view/frontend/web/js/view/payment/method-renderer/boltpay.js
+        window.boltConfig = <?php echo $block->getSettings(); ?>;
+        require([
+            'jquery',
+            'Magento_Customer/js/model/authentication-popup',
+            'Magento_Customer/js/customer-data',
+            'mage/validation/validation',
+            'mage/cookies'
+        ], function ($, authenticationPopup, customerData) {
+            var customer = customerData.get('customer');
+            var shoppingCart = customerData.get('cart');
+            var $qty = jQuery('#qty');
+            var max_qty = <?= $block->getProductQty(); ?>;
+            var isGuestCheckoutAllowed = <?= $block->isGuestCheckoutAllowed(); ?>;
+
+            /**
+             * Pure javascript ajax POST request function
+             *
+             * param string     url                 The request url
+             * param array      params              The URL query string parameters
+             * param callable   callback            The callback function to pass the response data to
+             *
+             * return XMLHttpRequest|ActiveXObject('Microsoft.XMLHTTP')
+             */
+            var ajaxPost = function(url, params, onSuccess, onError, async, headers) {
```

Comment:
Since jquery is required, please use $.post. We will do it in other places eventually.

Reply:
Actually I copied function **ajaxPost** from file replacejs.phtml. I believe in the future we can combine code for PPC and for cart page checkout in one place (as we did in WooC).
So it's better to refactor both button_product_page.phtml and replacejs.phtml in one PR otherwise we will increase the difference.

I propose to remove function ajaxPost in this asana https://app.asana.com/0/941920570700290/1155745644813038/f
Feel free to assign me this task.

Reply:
fair point